azext_iot/central/providers/device_provider.py

```diff
@@ -170,44 +171,91 @@ def get_device_credentials(
         return credentials
 
     def get_device_registration_info(
-        self, device_id, central_dns_suffix="azureiotcentral.com",
+        self,
+        device_id,
+        device_status: DeviceStatus,
+        central_dns_suffix="azureiotcentral.com",
     ):
+        dps_state = {}
         info = self._device_registration_info.get(device_id)
 
         if not info:
-            credentials = self.get_device_credentials(
-                device_id=device_id, central_dns_suffix=central_dns_suffix
+            device_info = self.get_device(device_id)
+            device_essential_info = central_services.device.device_populate_essential_info(
+                device_info, device_status
             )
-            id_scope = credentials["idScope"]
-            key = credentials["symmetricKey"]["primaryKey"]
-            dps_state = dps_global_service.get_registration_state(
-                id_scope=id_scope, key=key, device_id=device_id
+            if (
+                device_essential_info.get("deviceStatus")
+                is DeviceStatus.provisioned.value
+            ):
+                credentials = self.get_device_credentials(
+                    device_id=device_id, central_dns_suffix=central_dns_suffix
+                )
+                id_scope = credentials["idScope"]
+                key = credentials["symmetricKey"]["primaryKey"]
+                dps_state = dps_global_service.get_registration_state(
+                    id_scope=id_scope, key=key, device_id=device_id
+                )
+            dps_state = self.dps_populate_essential_info(
+                dps_state, device_essential_info.get("deviceStatus")
             )
-            central_info = self.get_device(device_id)
             info = {
                 "@device_id": device_id,
                 "dps_state": dps_state,
-                "central_info": central_info,
+                "device_info": device_essential_info,
             }
 
         self._device_registration_info[device_id] = info
 
         return info
 
-    def get_all_registration_info(self, central_dns_suffix="azureiotcentral.com"):
+    def dps_populate_essential_info(self, dps_info, device_status):
+        error = {
+            "provisioned": "None",
+            "registered": "Device it not yet provisioned.",
```

Comment:
nit: The `registered` value string has a period at the end `.`  It would be good to standardize on ending with a period or not.

Reply:
I'd suggest that cleaning up error messages is a P2 since I have a task in our backlog to clean up all error messages once we are ready to ship.

Let me know if this is something PR blocking and I can work with you to resolve